gym_electric_motor/envs/gym_eesm/cont_sc_eesm_env.py

```diff
@@ -0,0 +1,155 @@
+from gym_electric_motor.core import ElectricMotorEnvironment, ReferenceGenerator, RewardFunction, \
+    ElectricMotorVisualization
+from gym_electric_motor.physical_systems.physical_systems import ExternallyExcitedSynchronousMotorSystem, SynchronousMotorSystem
+from gym_electric_motor.visualization import MotorDashboard
+from gym_electric_motor.reference_generators import WienerProcessReferenceGenerator
+from gym_electric_motor import physical_systems as ps
+from gym_electric_motor.reward_functions import WeightedSumOfErrors
+from gym_electric_motor.utils import initialize
+from gym_electric_motor.constraints import SquaredConstraint, LimitConstraint
+
+
+class ContSpeedControlExternallyExcitedSynchronousMotorEnv(ElectricMotorEnvironment):
+    """
+    Description:
+        Environment to simulate a continuous control set speed controlled externally excited synchronous motor.
+
+    Key:
+        ``'Cont-SC-EESM-v0'``
+
+    Default Components:
+        - Supply: :py:class:`.IdealVoltageSupply`
+        - Converter: :py:class:`.ContB6BridgeConverter`
+        - Motor: :py:class:`.ExternallyExcitedSynchronousMotor`
+        - Load: :py:class:`.PolynomialStaticLoad`
+        - Ode-Solver: :py:class:`.ScipyOdeSolver`
+
+        - Reference Generator: :py:class:`.WienerProcessReferenceGenerator` *Reference Quantity:* ``'omega'``
+
+        - Reward Function: :py:class:`.WeightedSumOfErrors` reward_weights: ``'omega' = 1.0``
+
+        - Visualization: :py:class:`.MotorDashboard` current and action plots
+
+        - Constraints: :py:class:`.SquaredConstraint` on the currents  ``'i_sd', 'i_sq'``, :py:class:`.LimitConstraint` on the current  ``'i_e''``
+
+    State Variables:
+        ``['omega' , 'torque', 'i_sd', 'i_sq', 'i_a', 'i_b', 'i_c', 'i_e', 'u_sd', 'u_sq', 'u_a', 'u_b', 'u_c', 'u_e', 'u_sup']``
+
+    Referenced Variables:
+        ``['omega']``
+
+    Control Cycle Time:
+        tau = 1e-4 seconds
+
+    Observation Space:
+        Type: Tuple(State_Space, Reference_Space)
+
+    State Space:
+        Box(low=15 * [-1], high=15 * [1])
+
+    Reference Space:
+        Box(low=[-1, -1], high=[1, 1])
+
+    Action Space:
+        Box(low=[-1, -1, -1, -1], high=[1, 1, 1, 1])
```

Comment:
In the current control EESM env we had 3-dim Box constraints for i_d, i_q, i_e. Now, for the speed control env, we apparently have 4-dim Box constraints for i_abc, i_e? It should be 4 dimensional in both envs then, I assume?